### PR TITLE
Completed workshop3 refactor in develop branch, ready to merge into master branch after code review

### PR DESCRIPTION
**Pull request by frank-ye**

### CODE REVIEW

File: workshop_code/workshop_3/Workshop3.ipynb [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+     "end_time": "2018-10-20T04:33:19.920832Z",
+     "start_time": "2018-10-20T04:33:19.917660Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Do 60-20-20 train_test_split for training set, validation set, and test set\n",
+    "\n",
+    "def split(X, y):\n",
+    "    X_train, X_test, y_train, y_test = train_test_split(\n",
```
Comment (tiff-wang) [inline workshop_code/workshop_3/Workshop3.ipynb:195]: Maybe add a comment here saying that the second one would split the Test set into test and val?

File: workshop_code/workshop_3/Workshop3.ipynb [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+    "\n",
+    "degree = 10\n",
+    "for i in range(1,degree +1):\n",
+    "    X_poly = PF(df_energy_X, i)\n",
+    "    X_train, X_val, X_test, y_train, y_val, y_test = split(X_poly, df_energy_y)\n",
+    "    \n",
+    "    # Initialize linear reg model and fit data\n",
+    "    \n",
+    "    ### FILL IN ###\n",
+    "    print('Training set R^2 for degree {}: {}'.format(i, \n",
```
Comment (tiff-wang) [inline workshop_code/workshop_3/Workshop3.ipynb]: Fix indentation

File: workshop_code/workshop_3/Workshop3.ipynb [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-20T04:35:50.424919Z",
+     "start_time": "2018-10-20T04:35:50.382090Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# dot_data = tree.export_graphviz(clf, out_file=None, \n",
```
Comment (tiff-wang) [inline workshop_code/workshop_3/Workshop3.ipynb:497]: Add comment to what this does. 
why is it commented out?

File: workshop_code/workshop_3/Workshop3.ipynb [hunk 4]
```diff
@@ -0,0 +1,565 @@
...
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-20T04:33:19.394285Z",
+     "start_time": "2018-10-20T04:33:19.068177Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Method that takes in dataframe, and degree of polynomial\n",
```
Comment (tiff-wang) [inline workshop_code/workshop_3/Workshop3.ipynb]: I might just be really nit picky, but Python method comments are inside the function with ''' ''' :D 

https://realpython.com/documenting-python-code/#basics-of-commenting-code

**Review by tiff-wang — APPROVED** [review]

File: workshop_code/workshop_3/Workshop3.ipynb (hunk 1 above)
Comment (frank-ye) [reply workshop_code/workshop_3/Workshop3.ipynb:195]: sure, there are many cases like that, but I will explain it when we call the function to split the dataset into train, val, test sets. I do agree this could be a little cleaner tho :^)

File: workshop_code/workshop_3/Workshop3.ipynb (hunk 3 above)
Comment (frank-ye) [reply workshop_code/workshop_3/Workshop3.ipynb:497]: It doesn't work, I'm just leaving it there because the following image in the cell below it was generated using that code. The graphviz library is just difficult to set up
